### PR TITLE
[develop] Add o1js-stub to Mina repo in place of SnarkyJS

### PR DESCRIPTION
**Pull request by MartinMinkov**
# Summary

Addresses https://github.com/MinaProtocol/mina/issues/14466

🔗 o1js: https://github.com/o1-labs/o1js/pull/1318
🔗 o1js-bindings: https://github.com/o1-labs/o1js-bindings/pull/222

This PR aims to add support for building mina as a submodule of `o1js`. To do so, the following was done:
1. Removed the `snarkyjs` submodule
2. Added a `o1js_stub` module, that has the same dependencies as o1js. The goal with this module is to have a way for Mina CI to check that o1js can still be successfully compiled after changes are made. This gives a safety net to not totally break things if changes to the dependencies are made.
3. Adds the `kimchi` library from o1js-bindings into `src/lib/crypto`. This is to make sure that we can compile `o1js_stub` since we need the WASM dependencies generated from the library, as well as to support a longer term plan to create a o1 crypto repo that recombines the wasm bindings with o1caml-rust stubs.
4. Removes snarkyjs from nix and updates the Rust part of nix to reference the new kimchi library location
5. Removed codeowners entry for `snarkyjs`

### CODE REVIEW

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by MartinMinkov** [comment]
!ci-build-me

**Review by mitschabaude — APPROVED** [review]

**Review by bkase — APPROVED** [review]
I approve the CODEOWNERS change -- what tags me as a blocker on (I delegate the protocol and infra review to others even those GitHub will say it's ✅ )

File: src/lib/crypto/kimchi/wasm/src/pasta_fp_plonk_index.rs [hunk 1]
```diff
@@ -43,10 +43,8 @@ pub struct WasmPastaFpLookupTable {
 // below.
 impl From<WasmPastaFpLookupTable> for LookupTable<Fp> {
     fn from(wasm_lt: WasmPastaFpLookupTable) -> LookupTable<Fp> {
-        LookupTable {
-            id: wasm_lt.id.into(),
-            data: wasm_lt.data.0,
-        }
+        LookupTable::create(wasm_lt.id.into(), wasm_lt.data.0)
```
Comment (dannywillems) [inline src/lib/crypto/kimchi/wasm/src/pasta_fp_plonk_index.rs]: Revert. This comes from a reverted and buggy PR. Not compatible with proof-systems normally.

File: src/lib/crypto/kimchi/wasm/src/pasta_fp_plonk_index.rs [hunk 2]
```diff
@@ -190,36 +188,6 @@ pub fn caml_pasta_fp_plonk_index_domain_d8_size(index: &WasmPastaFpPlonkIndex) -
     index.0.cs.domain.d8.size() as i32
 }
 
-#[wasm_bindgen]
-pub fn caml_pasta_fp_plonk_index_decode(
```
Comment (dannywillems) [inline src/lib/crypto/kimchi/wasm/src/pasta_fp_plonk_index.rs]: Which commit did you use?

File: src/lib/crypto/kimchi/js/node_js/dune [hunk 3]
```diff
@@ -22,13 +21,13 @@
   ../../wasm/Cargo.lock
   (source_tree ../../wasm/src)
   (source_tree ../../wasm/.cargo/config)
-  (source_tree ../../../proof-systems))
```
Comment (dannywillems) [inline src/lib/crypto/kimchi/js/node_js/dune]: Which commit did you use? Is it supposed to be part of the commit message "Update kimchi library with latest develop changes"?

File: src/lib/crypto/kimchi/wasm/src/pasta_fq_plonk_index.rs [hunk 4]
```diff
@@ -40,8 +40,10 @@ pub struct WasmPastaFqLookupTable {
 
 impl From<WasmPastaFqLookupTable> for LookupTable<Fq> {
     fn from(wasm_lt: WasmPastaFqLookupTable) -> LookupTable<Fq> {
-        LookupTable::create(wasm_lt.id.into(), wasm_lt.data.0)
-            .expect("LookupTable -> WasmPastaFqLookupTable conversion must succeed")
+        LookupTable {
```
Comment (dannywillems) [inline src/lib/crypto/kimchi/wasm/src/pasta_fq_plonk_index.rs:43]: Ah, it is reverted here.

File: src/lib/crypto/kimchi/js/node_js/dune (hunk 3 above)
Comment (dannywillems) [reply src/lib/crypto/kimchi/js/node_js/dune]: It seems to be reverted in https://github.com/MinaProtocol/mina/pull/14649/commits/4c9d45ce46afd65ff18dc5f3ff8648cd87c9c697.

**Review by dannywillems — COMMENTED** [review]
Clean history! I like this, we should do more often. I have some comments though.

**Comment by dannywillems** [comment]
And same comment related to the CI and build than in https://github.com/MinaProtocol/mina/pull/14461

**Review by georgeee — APPROVED** [review]

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-build-me

**Review by dannywillems — APPROVED** [review]

**Pull request merged**